src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/SystemParameterService.java

```diff
@@ -0,0 +1,133 @@
+
+    public static final String INSTANCE_IDENTIFIER = "instanceIdentifier";
+    public static final String CENTRAL_SERVER_ADDRESS = "centralServerAddress";
+    public static final String AUTH_CERT_REG_URL = "authCertRegUrl";
+    public static final String DEFAULT_AUTH_CERT_REG_URL = "https://%{centralServerAddress}:4001/managementservice/";
+    public static final String CONF_HASH_ALGO_URI = "confHashAlgoUri";
+    public static final String DEFAULT_CONF_HASH_ALGO_URI = DigestMethod.SHA512;
+    public static final String CONF_SIGN_CERT_HASH_ALGO_URI = "confSignCertHashAlgoUri";
+    public static final String DEFAULT_CONF_SIGN_CERT_HASH_ALGO_URI = DigestMethod.SHA512;
+    public static final String SECURITY_SERVER_OWNERS_GROUP = "securityServerOwnersGroup";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP = "security-server-owners";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP_DESC = "Security server owners";
+    public static final String CONF_SIGN_DIGEST_ALGO_ID = "confSignDigestAlgoId";
+    public static final String DEFAULT_CONF_SIGN_DIGEST_ALGO_ID = CryptoUtils.SHA512_ID;
+    public static final String OCSP_FRESHNESS_SECONDS = "ocspFreshnessSeconds";
+    public static final Integer DEFAULT_OCSP_FRESHNESS_SECONDS = 3600;
+    public static final String TIME_STAMPING_INTERVAL_SECONDS = "timeStampingIntervalSeconds";
+    public static final Integer DEFAULT_TIME_STAMPING_INTERVAL_SECONDS = 60;
+    public static final String CONF_EXPIRE_INTERVAL_SECONDS = "confExpireIntervalSeconds";
+    public static final Integer DEFAULT_CONF_EXPIRE_INTERVAL_SECONDS = 600;
+    private static final String[] NODE_LOCAL_PARAMETERS = {CENTRAL_SERVER_ADDRESS};
+
+    private final SystemParameterRepository systemParameterRepository;
+    private final HAConfigStatus currentHaConfigStatus;
+
+    public String getParameterValue(String key, String defaultValue) {
+        Optional<SystemParameter> valueInDb;
+        log.debug("getParameterValue() - getting value for key:{} with default value:{}", key, defaultValue);
+        if (currentHaConfigStatus.isHaConfigured()
+                && isNodeLocalParameter(key)
+        ) {
+            valueInDb = systemParameterRepository.findByKeyAndHaNodeName(
+                    key,
+                    currentHaConfigStatus.getCurrentHaNodeName()
+            );
+        } else {
+            List<SystemParameter> systemParameters = systemParameterRepository.findByKey(key);
+            valueInDb = systemParameters.stream().findFirst();
+            assert 1 >= systemParameters.size();
+        }
+        if (valueInDb.isPresent()) {
+            String valueFromDb = valueInDb.get().getValue();
+            log.trace("getParameterValue found a {} SystemParameter value {}  ", key, valueFromDb);
+            return valueFromDb;
+        }
+        log.trace("getParameterValue returns given default value for key {}", key);
+        return defaultValue;
```

Comment:
Could be simplified (omitting logging) to 
`return valueInDb.map(SystemParameter::getValue).orElse(defaultValue)`;

Reply:
Brilliant!

Reply:
Fixed: 
79b76d0